### PR TITLE
[Chore] Bumps go version to 1.21 (latest stable)

### PR DESCRIPTION
**Pull request by gruyaume**
## Overview

This project currently uses Go 1.16 which is end of life. This PR bumps the go version to 1.21 which is the latest stable release. This was accomplished via:
```
go mod edit -go=1.21
go mod tidy
```

Similar PR's will be made in the other network functions after this one gets merged.

## Reference
- https://go.dev/blog/go1.21

### CODE REVIEW

**Comment by ajaythakurintel** [comment]
ok to test

**Comment by ajaythakurintel** [comment]
add to whitelist

**Comment by ajaythakurintel** [comment]
Have you tried to build the code with target "make docker-build"

**Comment by thakurajayL** [comment]
add to whitelist

**Comment by thakurajayL** [comment]
I see following error. Do you see the same ?

node0:~/5g/nrf$ go mod tidy
go: go.mod file indicates go 1.21, but maximum version supported by tidy is 1.19
node0:~/5g/nrf$
node0:~/5g/nrf$
node0:~/5g/nrf$ go version
go version go1.19 linux/amd64
node0:~/5g/nrf$

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/omec-project/nrf
 
-go 1.16
+go 1.21
```
Comment (thakurajayL) [inline go.mod:3]: go mod tidy did not work for me. Please see my general comment.

Comment (gruyaume) [reply go.mod:3]: Which version of go do you have installed on your system? Make sure it is 1.21.

```bash
guillaume@potiron:~/code/nrf-go$ sudo snap install go --channel=1.21/stable --classic
go (1.21/stable) 1.21.1 from Canonical✓ installed
guillaume@potiron:~/code/nrf-go$ go version
go version go1.21.1 linux/amd64
guillaume@potiron:~/code/nrf-go$ go mod tidy
...
```

Comment (thakurajayL) [reply go.mod:3]: I updated version. Now that error has dissappeared. Have you tried "make docker-build" ?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
 replace github.com/omec-project/nrf => ./
```
Comment (thakurajayL) [inline go.mod:5]: When I used your change I get errors in "make docker-build" but if i remove "replace github.com/omec-project/nrf => ./" then it works fine.

Comment (gruyaume) [reply go.mod]: Fixes now

File: go.mod (hunk 1 above)
Comment (gruyaume) [reply go.mod:3]: Yes, it's working now that I removed the "replace" line

**Review by thakurajayL — DISMISSED** [review]

**Comment by thakurajayL** [comment]
@dafATonf Can you help in knowing why FOSSA and LICENSE verification has not tirggered.

**Comment by dafATonf** [comment]
test this please

**Pull request merged**